bottlerocket/testsys/src/run_aws_k8s.rs

```diff
@@ -171,6 +172,15 @@ pub(crate) struct RunAwsK8s {
     /// The arn for the role that should be assumed by the agents.
     #[structopt(long)]
     assume_role: Option<String>,
+
+    /// The path to a TOML file containing custom userdata.
+    #[structopt(long, requires("custom_user_data_mode"))]
+    custom_user_data: Option<String>,
+
+    /// The way custom userdata should interact with the default userdata.
+    /// The possible values are `merge` and `replace`.
+    #[structopt(long, requires("custom_user_data"))]
+    custom_user_data_mode: Option<String>,
```

Comment:
same thing, avoid designing structures that are not self-validating. i.e. struct.field_a should not rely on struct.field_b being in a compatible state.